crates/sui-core/src/authority_active/gossip/mod.rs

```diff
@@ -307,38 +404,10 @@ where
                     }
                 },
                 digest = &mut queue.next() , if !queue.is_empty() => {
-                    let digest = digest.unwrap();
-                    if !self.state.database.effects_exists(&digest.transaction)? {
-                        // Download the certificate
-                        let response = self.client.handle_transaction_info_request(TransactionInfoRequest::from(digest.transaction)).await?;
-                        self.process_response(response).await?;
-                    }
+                    handler.handle_digest(self, digest.unwrap()).await?;
```

Comment:
Ok, cool so the point of all this is to make the function generic with respect to this call right? I presume eventually we will have a different one for full node sync.

Reply:
Yes, I'm working on that one, trying to break out smaller PRs whenever possible.